### ISSUE TITLE
Windows panics with text on the same line

### ISSUE THREAD

**Issue by jonathandturner** [kkawakam/rustyline#359]
We're running into a situation in Nushell where we're seeing rustyline panic in Windows. It seems to happen if there is text on the same line as the prompt when it's starting.

You can reproduce this in rustyline today. In the example called 'example', change: 

https://github.com/kkawakam/rustyline/blob/master/examples/example.rs#L111

To use `print!` instead of `println!`

Here's an example of the crash:
![image](https://user-images.githubusercontent.com/547158/78724289-3c70f800-7981-11ea-80c5-47f31bac6493.png)

I can keep looking for a fix, but you all will know the codebase better and may know what could be causing that.

**Comment by gwenn** [comment]
@jonathandturner Would you mind running the example with debug logs ?
```sh
> set RUST_LOG=rustyline=debug
> cargo run --example example 2> debug.log
```
And post the "initial cursor location" when the error occurs.

I tried to reproduce the crash on Windows 10 without success.

**Comment by jonathandturner** [comment]
```
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `target\debug\examples\example.exe`
[2020-04-08T16:56:45Z DEBUG rustyline::tty::windows] ANSI colors already enabled
[2020-04-08T16:56:45Z DEBUG rustyline::edit] old layout: Layout { prompt_size: Position { col: 0, row: 0 }, default_prompt: false, cursor: Position { col:
 0, row: 0 }, end: Position { col: 0, row: 0 } }
[2020-04-08T16:56:45Z DEBUG rustyline::edit] new layout: Layout { prompt_size: Position { col: 3, row: 0 }, default_prompt: true, cursor: Position { col:
3, row: 0 }, end: Position { col: 3, row: 0 } }
1> [2020-04-08T16:56:47Z DEBUG rustyline::keymap] Emacs command: AcceptLine
[2020-04-08T16:56:47Z DEBUG rustyline::undo] Changeset::begin
[2020-04-08T16:56:47Z DEBUG rustyline::undo] Changeset::end

[2020-04-08T16:56:47Z DEBUG rustyline::tty::windows] ANSI colors already enabled
Line: [2020-04-08T16:56:47Z DEBUG rustyline::tty::windows] initial cursor location: 6, 38

Error: Io(Os { code: 87, kind: Other, message: "The parameter is incorrect." })
```

I noticed that you might need to be in Windows Terminal and not the old cmd to repro the issue.

**Comment by gwenn** [comment]
@jonathandturner Maybe `38 + 1` is not a valid row ?
Tomorrow, I will try again and if I still can't reproduce the crash I will merge your PR.
Regards.

**Comment by gwenn** [comment]
@jonathandturner I guess we should use `println!("")` only while calling `move_cursor_at_leftmost`.
See #361.
I will do some tests tomorrow.

**Comment by gwenn** [comment]
Sorry,
the windows 10 revision on my desktop is not compatible with windows terminal.
So I can't check that #361 really works.

**Comment by jonathandturner** [comment]
I'll take a look

**Comment by jonathandturner** [comment]
@gwenn - unfortunately #361 doesn't fix this issue.

**Comment by gwenn** [comment]
@jonathandturner Silly me !
```
Error: Io(Os { code: 87, kind: Other, message: "The parameter is incorrect." })
```
Kind is `Other`...

**Comment by jonathandturner** [comment]
@gwenn that fixed it

**Comment by gwenn** [comment]
Should be fixed by version 6.1.1
